Makefile

```diff
@@ -77,7 +77,7 @@ test-in-docker:
 	# Privileged needed for docker-in-docker so integ tests pass
 	docker run -v "$(shell pwd):/go/src/github.com/aws/amazon-ecs-agent" --privileged "amazon/amazon-ecs-agent-test:make"
 
-run-functional-tests: test-registry
+run-functional-tests: docker test-registry
```

Comment:
Unconditionally building a newer version of the Agent won't necessarily be preferable. For example, if you had test automation that ran the latest sets of tests against existing agents / AMIs, this might break that.

If that existed, it could make more sense to only run that target if such an image doesn't yet exist. There's also an environment variable that can change which image the tests run against, which could either make this a waste of time or incorrect depending on your point of view.